docs/src/docs/common-pitfalls.md

```diff
@@ -0,0 +1,76 @@
+
+First, we need to understand that a Slinkity build consists of multiple steps:
+
+1. Processing your site with 11ty and dumping the result into a temporary build folder (e.g., `.11ty-build-DcNVBN`).
+2. Processing the temporary build folder with Vite to bundle assets on the pages that reference them.
+3. Writing the temporary build folder to your 11ty output directory (e.g., `_site`)
+4. Deleting the temporary build folder.
+
+Vite is very strict about where it wants you to place your static assets—in a [dedicated `public` folder](https://vitejs.dev/guide/assets.html#the-public-directory). This allows it to distinguish static assets from ones that are referenced by your pages (like stylesheets) and to bundle the referenced assets while leaving the static ones alone. Any static asset that isn't referenced by a page and isn't in the `public` directory will be removed from the final build output. For this reason, you may find that certain files disappear from production builds, even though they were technically processed correctly by 11ty.
+
+Below are some scenarios where you may encounter this pitfall:
+
+- Self-hosted font files
+- Static images (both processed and passthrough-copied)
+- Miscellaneous static files (`sitemap.xml`, `robots.txt`, and the like)
+
+Let's look at an example.
+
+### The old approach (11ty)
+
+Suppose you're using 11ty to auto-generate a `sitemap.xml` for your site. In 11ty, you'd create this file using your chosen templating language extension and set a permalink in its front matter relative to the output directory, like so:
+
+{% raw %}
+```liquid
+{% comment %}src/sitemap.liquid{% endcomment %}
+---
+permalink: /sitemap.xml
+---
+```
+{% endraw %}
+
+When 11ty builds your site, it'll process this template file and spit out a file named `_site/sitemap.xml`.
+
+Unfortunately, this won't work in Slinkity. In the first step, the sitemap will get written to `.11ty-build-DcNVBN/sitemap.xml`. In the follow-up steps, Vite will process this temporary build directory, see that `sitemap.xml` is not referenced by any other file and isn't in the dedicated `public` folder, and exclude it from the final output that it writes to `_site`.
+
+### The new approach (Slinkity)
+
+To fix this problem, you need to prefix the static file's permalink with `/public`.
+
+So instead of this:
+
+{% raw %}
+```liquid
+{% comment %}src/sitemap.liquid{% endcomment %}
+---
+permalink: /sitemap.xml
+---
+```
+{% endraw %}
+
+You would do this:
+
+{% raw %}
+```liquid
+{% comment %}src/sitemap.liquid{% endcomment %}
+---
+permalink: /public/sitemap.xml
+---
+```
+{% endraw %}
+
+Now, this happens:
+
+1. 11ty processes the template and writes it to a `public` folder nested inside the temporary build folder: `.11ty-build-DcNVBN/public/sitemap.xml`.
+2. Vite sees the `public` folder and copies it into your final build output directory, giving you `_site/public/sitemap.xml`.
```

Comment:
```suggestion
2. Vite sees the `public` folder and copies it into your final build output directory, giving you `_site/sitemap.xml`. Note the nested `/public` disappears in the final build output!
```

☝️ Fun fact: `public` is a disappearing directory!

Reply:
Oh, I still see it in my build output folder. Is that a bug?

![image](https://user-images.githubusercontent.com/19352442/148543433-2e2a3a6d-9fb3-4904-a7dc-c19a25a27c7c.png)

It still works because browsers just ignore the public bit, so a request to `/sitemap.xml` goes through successfully (in fact, attempting to access `/public/sitemap.xml` directly gives a 404.)